### ISSUE TITLE
Search bar

### ISSUE THREAD

**Issue by chris1000** [Rocketseat/gatsby-themes#61]
Was trying to figure out if there was a way to create a search bar in the side bar that doesn't use Algolia search for example (trying to build the site as cheap as possible)

**What would you like to be added**:

Simple search bar in the side bar and displays results in the same element.

**Why is this needed**:

Would make the docs easier to navigate.

**Is your enhancement request related to a problem? Please describe.**

No.

**Additional context**

Thank you, this project is awesome!

**Comment by diego3g** [comment]
Algolia is free for open-source projects. If it's not open-source, there is no easy way to achieve that with a static website.

**Comment by chris1000** [comment]
> Algolia is free for open-source projects. If it's not open-source, there is no easy way to achieve that with a static website.

Would it be possible to add a JS-Search option? Similar to how DocZ does it? I'll look into Algolia, they have a doc option that seems like it could work well.

**Comment by jpedroschmitz** [comment]
Hey @chris1000 👋🏻

As @diego3g has mentioned, you can use Algolia if your project will be Open Source. It is perfect for technical documentation, you can find their website [in this link](https://docsearch.algolia.com/).

If you want to use your own search, with a Gatsby Plugin, I'd recommend you use something like [`elasticlunr-search`](https://gatsbyjs.com/plugins/@gatsby-contrib/gatsby-plugin-elasticlunr-search/) or [`local-search`](https://gatsbyjs.com/plugins/gatsby-plugin-local-search/), that supports `lunr` and `FlexSearch`.

There's another option also, `tinysearch`, which is a great solution for static websites. But I don't have any tutorials on how to integrate it with React/Gatsby.

**Comment by chrislevinecentric** [comment]
Updated: I got approved for the DocsSearch however how can I shadow the sidebar to add the search bar in? I tried creating a file at src\@rocketseat\gatsby-theme-docs\components\Sidebar\index.js and copy the contents from the original file but I get missing file errors and I tried using the useSidebar hook and I'm getting errors as well, any help would be awesome! Seriously awesome project, thank you for the help!
